src/olympia/files/models.py

```diff
@@ -378,40 +378,53 @@ def get_localepicker(self):
     def webext_permissions(self):
         """Return permissions that should be displayed, with descriptions, in
         defined order:
-        1) match all urls (e.g. <all-urls>)
-        2) known permissions, in constants order (alphabetically),
-        3) match urls for sites
+        1) Either the match all permission, if present (e.g. <all-urls>), or
+           match urls for sites (<all-urls> takes preference over match urls)
+        2) nativeMessaging permission, if present
+        3) other known permissions in alphabetical order
         """
         knowns = list(WebextPermissionDescription.objects.filter(
             name__in=self.webext_permissions_list).iterator())
-
-        match_all, urls = [], []
+        # Move nativeMessaging permission to front of the list, if present
+        for index, perm in enumerate(knowns):
```

Comment:
Best we can do is add a check for the permission being in `webext_permissions_list` and then only enumerate the knowns and re-sort if it's present.  E.g.
```
        knowns = list(WebextPermissionDescription.objects.filter(
            name__in=self.webext_permissions_list).iterator())

        urls = []
        match_url = None
        for name in self.webext_permissions_list:
            if re.match(WebextPermissionDescription.MATCH_ALL_REGEX, name):
                match_url = WebextPermissionDescription.ALL_URLS_PERMISSION
            elif name == WebextPermission.NATIVE_MESSAGING_NAME:
                # Move nativeMessaging to front of the list, if present
                for index, perm in enumerate(knowns):
                    if perm.name == WebextPermission.NATIVE_MESSAGING_NAME:
                        knowns.pop(index)
                        knowns.insert(0, perm)
                        break
            elif '//' in name:
                # Filter out match urls so we can group them.
                urls.append(name)
            # Other strings are unknown permissions we don't care about
... match_url code ...
```